### ISSUE TITLE
Core debugger do not generate results

### ISSUE THREAD

**Issue by ralvarep** [fp7-netide/Tools#3]
Hi,

I am testing core debugger with the latest Engine commit and it do not generate any results in _results.txt_ and _results.pcap_. It creates empty files.

```
netide@netide-VirtualBox:~/NetIDE/Tools/debugger/Core$ python debugger.py 
 [*] Waiting for logs. To exit press CTRL+C
^C

netide@netide-VirtualBox:~/NetIDE/Tools/debugger/Core$ ls -alh
total 108K
drwxrwxr-x 2 netide netide 4,0K jul 30  2015 .
drwxrwxrwx 4 netide netide 4,0K feb 24 14:51 ..
-rw-rw-r-- 1 netide netide 4,2K feb 24 14:51 debugger.py
-rw-rw-r-- 1 netide netide  11K mar  9 11:01 demo1.pcap
-rw-rw-r-- 1 netide netide  26K mar  9 12:03 demo2.pcap
-rw-rw-r-- 1 netide netide  968 feb 24 14:51 netide.la
-rw-rw-r-- 1 netide netide  31K feb 24 14:51 netide.so
-rw-rw-r-- 1 netide netide  12K feb 24 14:51 packet-user_encap.c
-rw-rw-r-- 1 netide netide 1,3K feb 24 14:51 README.md
-rw-rw-r-- 1 netide netide    0 abr  8 13:55 results.pcap
-rw-rw-r-- 1 netide netide    0 abr  8 13:55 results.txt
```

LogPub bundle starts correctly:

```
netide-core>la | grep logpub
72 | Active   |  80 | 1.1.0.SNAPSHOT  | core.logpub Bundle
```

Thanks in advance!

**Comment by sergiotamu** [comment]
Hello @ralvarep ,

I have been testing and there is an error in the LogPub bundle. For that reason you cannot see the messages in the Logger and the debugger does not generate any outcome.

This is the error:
Exception in thread "LogPub Receive Loop" java.lang.NullPointerException
    at eu.netide.core.logpub.LogPub.run(LogPub.java:87)
    at java.lang.Thread.run(Thread.java:745)

Maybe @KevinPhemius can help us with this issue.

**Comment by KevinPhemius** [comment]
Hi,

there seems to be some issue I can't pinpoint.
When I build the code for the LogPub from the LogPubSub branch, the test hangs here
<pre>
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running TestLogPub
Configuring TestNG with: TestNG652Configurator
14:15:58.027 [main] INFO  eu.netide.core.logpub.LogPub - LogPub Constructor.
14:15:58.042 [main] INFO  eu.netide.core.logpub.LogPub - LogPub Start().
14:15:58.126 [LogPub Receive Loop] INFO  eu.netide.core.logpub.LogPub - LogPub started.
Message from backend test
14:15:58.281 [main] DEBUG eu.netide.core.logpub.LogPub - Received message to 2_all from backend:[ 2_all, 0_b1, 010100040000000800000017000000000000002A00000000 ]
14:15:58.358 [LogPub Receive Loop] INFO  eu.netide.core.logpub.LogPub - Listening PUB queue on port 0
14:15:58.408 [LogPub Receive Loop] INFO  eu.netide.core.logpub.LogPub - Listening SUB queue on port 0
14:15:58.413 [LogPub Receive Loop] INFO  eu.netide.core.logpub.LogPub - Control queue on address: inproc://LogPubControl
^C
</pre>

When I build the code from *master* (exact same LogPub code, except the variable names), it hangs there
<pre>
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running TestLogPub
Configuring TestNG with: TestNG652Configurator
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Message from backend test
^C
</pre>

It didn't have this behavior before. Does the NetIP library changes could have this effect?

**Comment by KevinPhemius** [comment]
The pom.xml was missing this line:
<pre>
<Bundle-Activator>eu.netide.core.logpub.Activator</Bundle-Activator>
</pre>

**Comment by ralvarep** [comment]
Problem fixed in fp7-netide/Engine#91. I close this issue.
